### PR TITLE
fix: some adapters don't support visibility

### PR DESCRIPTION
**Pull request by dkarlovi**

### CODE REVIEW

File: src/Flysystem/StreamCommand/StreamOpenCommand.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public static function run(
 
             $current->alwaysAppend = 'a' === $mode[0];
             if (is_resource($current->handle) && !$current->alwaysAppend) {
-                rewind($current->handle);
+                @rewind($current->handle);
```
Comment (dkarlovi) [inline src/Flysystem/StreamCommand/StreamOpenCommand.php:71]: There might be a better way to handle this, but here a warning is dispatched for me:

```
Warning: rewind(): Stream does not support seeking  
```

Maybe a check would be better here?

**Comment by m2mtech** [comment]
Dear @dkarlovi 
these changes go haywire with the unit tests.
Could it be you should provide a default mode instead of 0?
What happens with adapters where there should be an error when there is no visibility?
Thanks for your contribution!

**Comment by dkarlovi** [comment]
@m2mtech for example, [Flysystem's Azure Storage adapter just throws an exception](https://github.com/thephpleague/flysystem/blob/aa7119c502873324860004e0bbc86281d98028b7/src/AzureBlobStorage/AzureBlobStorageAdapter.php#L226).

**Comment by m2mtech** [comment]
@dkarlovi Sorry I did not express myself clearly enough.

If you have a look at [LocalFlysystemAdapter](https://github.com/thephpleague/flysystem/blob/aa7119c502873324860004e0bbc86281d98028b7/src/Local/LocalFilesystemAdapter.php#L342), it throws an exception when there is a problem getting the visibility. How to you differentiate that there is an exception because of a problem from that the adapter throws the exception because it does not implement a functionality? 

Personally I think that when Azure does not support visibility, everything on Azure - concerning flysystem - is public and the adapter return PUBLIC, instead of throwing an exception. But that is philosophy we can argue about...

So the question is should we ignore all errors concerning visibility, send a PR to the Azure adapter to return PUBLIC instead of the exception?

**Comment by dkarlovi** [comment]
@m2mtech I think it's more complex than that. Azure will in some cases support visibility, but by default it will not. But in that case private visibility is actually a better choice here maybe?

**Comment by m2mtech** [comment]
Dear @dkarlovi 
with the latest commit you can optionally allow to ignore visibility errors, e.g.:

```php
FlysystemStreamWrapper::register('fly', $filesystem, [
    FlysystemStreamWrapper::IGNORE_VISIBILITY_ERROS => true,
]);
```

Could you please check if this solves your issue?

**Pull request closed without merging**